src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java

```diff
@@ -94,17 +94,52 @@ private void populateNewVendorCorporateAddressAttributes(PaymentWorksVendor stgN
+        PaymentWorksRemittanceAddressDTO pmwRemittanceAddressDTO = null;
         if (singleRemittanceAddressExists(pmwRequestingCompanyDTO)) {
-            PaymentWorksRemittanceAddressDTO pmwRemittanceAddressDTO = pmwRequestingCompanyDTO.getRemittance_addresses().getRemittance_address().get(0);
-            stgNewVendor.setRemittanceAddressStreet1(pmwRemittanceAddressDTO.getStreet1());
-            stgNewVendor.setRemittanceAddressStreet2(pmwRemittanceAddressDTO.getStreet2());
-            stgNewVendor.setRemittanceAddressCity(pmwRemittanceAddressDTO.getCity());
-            stgNewVendor.setRemittanceAddressState(pmwRemittanceAddressDTO.getState());
-            stgNewVendor.setRemittanceAddressCountry(pmwRemittanceAddressDTO.getCountry());
-            stgNewVendor.setRemittanceAddressZipCode(pmwRemittanceAddressDTO.getZipcode());
-            stgNewVendor.setRemittanceAddressValidated(pmwRemittanceAddressDTO.getValidated());
+            LOG.info("populateNewVendorRemittanceAddressAttributes: Single remit address found.");
+            pmwRemittanceAddressDTO = pmwRequestingCompanyDTO.getRemittance_addresses().getRemittance_address().get(0);
+            populateEachNewVendorRemitAddressAttribute(stgNewVendor, pmwRemittanceAddressDTO);
             populateNewVendorBankAccountAttributes(stgNewVendor, pmwRemittanceAddressDTO);
         }
+        else {
+            pmwRemittanceAddressDTO = determineRemitAddressToUse(pmwRequestingCompanyDTO);
+            if (ObjectUtils.isNotNull(pmwRemittanceAddressDTO)) {
+                LOG.info("populateNewVendorRemittanceAddressAttributes: Remit addresses found to use from MULTIPLE remit addresses.");
+                populateEachNewVendorRemitAddressAttribute(stgNewVendor, pmwRemittanceAddressDTO);
+                populateNewVendorBankAccountAttributes(stgNewVendor, pmwRemittanceAddressDTO);
+            }
+            else {
+                LOG.info("populateNewVendorRemittanceAddressAttributes: No remit addresses found.");
+            }
+        }
+    }
+    
+    private PaymentWorksRemittanceAddressDTO determineRemitAddressToUse(PaymentWorksRequestingCompanyDTO pmwRequestingCompanyDTO) {
+        PaymentWorksRemittanceAddressDTO pmwRemittanceAddressDTO = null;
+        if (multipleRemittanceAddressesExist(pmwRequestingCompanyDTO)) {
+            List<PaymentWorksRemittanceAddressDTO> pmwRemittanceAddressesDTO = pmwRequestingCompanyDTO.getRemittance_addresses().getRemittance_address();
+            int index = 0;
+            boolean haveNotFoundAddressToUse = true;
+            
+            while (haveNotFoundAddressToUse && (index < pmwRemittanceAddressesDTO.size())) {
+                pmwRemittanceAddressDTO = pmwRemittanceAddressesDTO.get(index);
+                if (bankAccountDataExists(pmwRemittanceAddressDTO)) {
+                    haveNotFoundAddressToUse = false;
+                }
+                index++;
+            }
```

Comment:
This loop's logic could be represented more cleanly through the use of stream-based filtering. Given the urgency of this fix, though, I am okay with this loop going out as-is if there is not enough time to refactor it at the moment.

Reply:
Due to the fact that the stream would be based on an XML object class that is going to be refactored at a later date anyway by a JIRA presently in the PMW backlog, I would prefer to defer the use of streams at the moment.